### PR TITLE
Biopar vito porting to APEX

### PR DESCRIPTION
**Pull request by Pratichhya**
The UDP build, published within the Copernicus Data Space Ecosystem, has been modified to include only the JSON file, with the generate.py script excluded.

Additionally, as per the document's requirements, the repository link is not publicly available and therefore has not been included.

### CODE REVIEW

**Comment by Pratichhya** [comment]
However, the actions failed, suggesting a need for linting. I have corrected the linting issues but am still unable to identify the root cause. Could you please suggest the necessary changes?

File: openeo_udp/biopar/biopar.json [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        {
+            "href": "https://marketplace-portal.dataspace.copernicus.eu/catalogue/app-details/21",
+            "rel": "about",
+            "title": "openEO Algorithm Plaza"
+        }
+    ],
+    "parameters": [
+        {
+            "description": "Left-closed temporal interval, i.e. an array with exactly two elements:\n\n1. The first element is the start of the temporal interval. The specified instance in time is **included** in the interval.\n2. The second element is the end of the temporal interval. The specified instance in time is **excluded** from the interval.\n\nThe specified temporal strings follow [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339.html). Also supports open intervals by setting one of the boundaries to `null`, but never both.",
+            "name": "date",
```
[the remark this answers is not in the document]
Comment (Pratichhya) [reply openeo_udp/biopar/biopar.json]: I assume here you meant temporal_extent?

**Comment by Pratichhya** [comment]
made couple of basic changes.
FYI: the PR doesn't include example_output link yet.

File: openeo_udp/biopar/biopar.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (soxofaan) [reply openeo_udp/biopar/biopar.json]: indeed

File: openeo_udp/biopar/biopar.json [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    "id": "BIOPAR",
```
Comment (soxofaan) [inline openeo_udp/biopar/biopar.json:3]: Unit tests fail on upper case "BIOPAR". 
openeo processes are all lower case by convention

File: benchmark_scenarios/biopar.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
+[
+    {
+        "id": "Biopar_fapar",
+        "type": "openeo",
+        "description": "Calculating FAPAR using Biopar service available in Copernicus Data Space Ecosystem",
+        "backend": "openeofed.dataspace.copernicus.eu",
+        "process_graph": {
+            "BIOPAR": {
+                "process_id": "BIOPAR",
```
Comment (soxofaan) [inline benchmark_scenarios/biopar.json:9]: As noted, openeo processes should be lower case (by convention)

**Comment by Pratichhya** [comment]
@soxofaan, I assume the PR is ready. Please let me know if I missed anything

**Comment by soxofaan** [comment]
The test still fail, so there must still be something wrong (haven't looked into it myself yet)

File: algorithm_catalog/biopar.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        ],
+        "license": "other"
+    },
+    "linkTemplates": [],
+    "links": [
+        {
+            "rel": "openeo-process",
+            "type": "application/json",
+            "title": "openEO Process Definition",
+            "href": "https://raw.githubusercontent.com/ESA-APEx/apex_algorithms/refs/heads/biopar_vito/openeo_udp/biopar/biopar.jsonBIOPAR"
```
[the remark this answers is not in the document]
Comment (Pratichhya) [reply algorithm_catalog/biopar.json]: sorry, yes, it was unintended. Will correct it right away.

**Comment by Pratichhya** [comment]
> The test still fail, so there must still be something wrong (haven't looked into it myself yet)

as a layman thought, the failure seems to occure in worldcereal interence josn file:HTTPError: 404 Client Error: Not Found for url: https://raw.githubusercontent.com/ESA-APEX/apex_algorithms/main/openeo_udp/worldcereal_inference.json

**Comment by soxofaan** [comment]
> ah I would merge the main branch in this feature branch then

I guess I wasn't clear here: you should merge the main branch in your feature branch to resolve the failing unit test

**Comment by Pratichhya** [comment]
> I guess I wasn't clear here: you should merge the main branch in your feature branch to resolve the failing unit test

ahh haha, no I was too excited that I only saw the merge

**Comment by soxofaan** [comment]
```
>       assert "geometry" in data
E       AssertionError: assert 'geometry' in {'conformsTo': ['http://www.opengis.net/spec/ogcapi-records-1/1.0/req/record-core'], 'id': 'biopar', 'linkTemplates': ...ex-examples/biopar/biopar-example.tif', 'rel': 'example', 'title': 'Example output', 'type': 'application/json'}], ...}
```


"geometry" is a required field in algorithm records
also see https://github.com/ESA-APEx/apex_algorithms/commit/cd199d370caff390446ab903f76f445e9e56da1b

**Comment by Pratichhya** [comment]
@soxofaan finally its green, thank you very much

File: benchmark_scenarios/biopar.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+[
+    {
+        "id": "biopar",
+        "type": "openeo",
+        "description": "Calculating FAPAR using Biopar service available in Copernicus Data Space Ecosystem",
+        "backend": "openeofed.dataspace.copernicus.eu",
+        "process_graph": {
+            "biopar1": {
+                "process_id": "biopar",
+                "namespace": "https://raw.githubusercontent.com/ESA-APEx/apex_algorithms/refs/heads/biopar_vito/openeo_udp/biopar/biopar.json",
```
Comment (soxofaan) [inline benchmark_scenarios/biopar.json:10]: This URL will break once the feature branch `biopar_vito` is removed

File: openeo_udp/biopar/biopar.json [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    "id": "biopar",
+    "links": [
+        {
+            "href": "https://raw.githubusercontent.com/ESA-APEx/apex_algorithms/refs/heads/biopar_vito/openeo_udp/biopar/biopar.json",
```
Comment (soxofaan) [inline openeo_udp/biopar/biopar.json:6]: This URL will also break once the feature branch biopar_vito is removed

Comment (soxofaan) [reply openeo_udp/biopar/biopar.json:6]: change it to
https://raw.githubusercontent.com/ESA-APEx/apex_algorithms/3b5a011a90f4a3050ff8fdf69ca5bc2fd1535881/openeo_udp/biopar/biopar.json

File: benchmark_scenarios/biopar.json (hunk 5 above)
Comment (soxofaan) [reply benchmark_scenarios/biopar.json:10]: change it to https://raw.githubusercontent.com/ESA-APEx/apex_algorithms/3b5a011a90f4a3050ff8fdf69ca5bc2fd1535881/openeo_udp/biopar/biopar.json

File: openeo_udp/biopar/biopar.json [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+            "rel": "canonical",
+            "title": "Public URL for user-defined process 'BIOPAR'"
```
Comment (soxofaan) [inline openeo_udp/biopar/biopar.json:8]:
```suggestion
            "title": "Public URL for user-defined process 'biopar'"
```

File: algorithm_catalog/biopar.json [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        ],
+        "license": "other"
+    },
+    "linkTemplates": [],
+    "links": [
+        {
+            "rel": "openeo-process",
+            "type": "application/json",
+            "title": "openEO Process Definition",
+            "href": "https://raw.githubusercontent.com/ESA-APEx/apex_algorithms/refs/heads/biopar_vito/openeo_udp/biopar/biopar.json"
```
Comment (soxofaan) [inline algorithm_catalog/biopar.json:103]: This URL will break once the feature branch biopar_vito is removed

change it to https://raw.githubusercontent.com/ESA-APEx/apex_algorithms/3b5a011a90f4a3050ff8fdf69ca5bc2fd1535881/openeo_udp/biopar/biopar.json

File: openeo_udp/biopar/biopar.json [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+            }
+        },
+        {
+            "default": "FAPAR",
+            "description": "BIOPAR type [FAPAR,LAI,FCOVER,CCC,CWC]",
+            "name": "biopar_type",
+            "optional": true,
+            "schema": {
+                "type": "string"
+            }
```
Comment (soxofaan) [inline openeo_udp/biopar/biopar.json:76]: This parameter seems to be a enum, with predefined possible values. If that is the case, the schema should reflect that properly.
For example see https://github.com/Open-EO/openeo-processes/blob/6141771129d58d2a292db3b91d80c812c49f4e52/add_dimension.json#L37-L50  for an example


Also note that if you use the python client to build the UDP, you can do that with the `values` argument of `Parameter.string` (https://open-eo.github.io/openeo-python-client/api.html#openeo.api.process.Parameter.string)

**Pull request merged**